src/web/components/ClientSideTokenGeneration/CstgAddDomainDialog.tsx

```diff
@@ -28,49 +35,67 @@ function CstgAddDomainDialog({
       setError('root.serverError', {
         type: '400',
-        message: 'The domain(s) you have entered already exist.',
+        message: 'The domains entered already exist.',
       });
     } else {
-      const allValid = uniqueDomains.every((newDomainName) => {
-        return isValidDomain(newDomainName);
+      const invalidDomains: string[] = [];
+      let allValid = true;
+      uniqueDomains.forEach((newDomain) => {
+        if (!isValidDomain(newDomain)) {
+          invalidDomains.push(newDomain);
+          allValid = false;
+        }
       });
       if (!allValid) {
         setError('root.serverError', {
           type: '400',
-          message: 'At least one domain you have entered is invalid, please try again.',
+          message: `The domains entered are invalid: ${formatStringsWithSeparator(invalidDomains)}`,
         });
       } else {
         // if all are valid but there are some non top-level domains, we make sure every domain is top-level
-        uniqueDomains.forEach((newDomainName, index) => {
-          uniqueDomains[index] = extractTopLevelDomain(newDomainName);
+        uniqueDomains.forEach((newDomain, index) => {
+          uniqueDomains[index] = extractTopLevelDomain(newDomain);
         });
-        await onAddDomains(uniqueDomains);
+        // filter for uniqueness (e.g. 2 different domains entered could have the same top-level domain)
+        const dedupedDomains = deduplicateStrings(uniqueDomains);
+        await onAddDomains(dedupedDomains, deleteExistingList);
       }
     }
   };
 
+  const onClickCheckbox = () => {
+    setDeleteExistingList(!deleteExistingList);
+  };
+
   return (
     <div className='add-domain-dialog'>
-      <Dialog title='Add Domain(s)' closeButtonText='Cancel' open onOpenChange={onOpenChange}>
+      <Dialog title='Add Domains' closeButtonText='Cancel' open onOpenChange={onOpenChange}>
         <RootFormErrors fieldErrors={errors} />
         <FormProvider {...formMethods}>
           <form onSubmit={handleSubmit(onSubmit)}>
-            You may enter a single domain or enter domains as a comma separated list.
-            <TextInput
-              inputName='newDomainNames'
-              label='Domain Name(s)'
-              rules={{ required: 'Please specify domain name(s).' }}
+            Add one or more domains.
+            <div className='checkbox-container'>
+              <StyledCheckbox
+                className='checkbox'
+                onClick={onClickCheckbox}
+                checked={deleteExistingList}
+              />
+              Replace all existing domains with the new ones.
+            </div>
+            <MultilineTextInput
+              inputName='newDomains'
+              label='Domains'
+              rules={{ required: 'Please specify domains.' }}
             />
             <div className='form-footer'>
               <button type='submit' className='primary-button'>
-                Add domains
+                Add Domains
```

Comment:
You could update the text of this button if `deleteExistingList` is checked - maybe "Replace Domains"?

Reply:
Good idea! Changed